### PR TITLE
Add Airflow variable for turning off provider DAG catchup

### LINKED ISSUE #1284: Turn `catchup` off for dated DAGs for local testing
## Problem

`catchup` is True for all of our dated DAGs. When you're testing locally, this can be very annoying because each time you run a dated DAG in a freshly built env, it will automatically start backfilling. This can be frustrating when you're trying to do manual runs on specific dates, for example.

I often find myself turning this off locally with a code change, but it must be re-applied when switching branches and you must be careful not to commit it.

## Description

One quick idea is to add an Airflow variable to control `catchup` for dated DAGs, and default it to True. Then we could set this in our local `.env` to persist this change when switching branches/rebuilding.

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Fixes #1284 by @stacimc 

## Description

Adds an Airflow variable called `CATCHUP_ENABLED` which can be used to override the default catchup settings for provider DAGs.

In production, `catchup` is enabled for all dated DAGs to allow for backfilling -- but this can be very annoying in local environments, because enabling a dated DAG will automatically kick off a very long backfill which you might not want.

Using an Airflow variable allows devs to easily toggle this on and off in local testing, and even allows us to do so in production should the need ever arise. If you want to persist the setting across `recreate`s of the environment, you can set `AIRFLOW_VAR_CATCHUP_ENABLED=false` in your `.env`. 

### Test default
* Make no changes to the Variables to begin with
* Select a few dated DAGs (eg metropolitan, flickr) and ensure that `catchup` is `True` by checking the Dag Details tab in the UI

### Test Airflow UI
* Create a variable through the Admin > Variables UI called "CATCHUP_ENABLED", and set it to 'false'
* Ensure that the dated DAGs now have catchup disabled
* Set the variable to 'true' and ensure catchup is enabled again

### Test env variable
* Add `AIRFLOW_VAR_CATCHUP_ENABLED=false` to your `catalog/.env` and run `just catalog/up`
* Ensure that the dated DAGs now have catchup disabled

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by krysal — APPROVED** [review]
Great. I'll go ahead and disable catchups locally as well. I believe it is worth to also add the variable to the `env.template` file.

**Comment by stacimc** [comment]
Oh, interesting -- the DAG docs generator determines whether a DAG is dated by checking `dag.catchup`, so by turning catchup off by default in the `env.template` it now thinks that none of the DAGs are dated! Taking a look.

**Comment by stacimc** [comment]
Fix for the DAG docs generator pushed. I'm glad you suggested defaulting the variable, @krysal, or I would not have caught that!

**Comment by stacimc** [comment]
Drafting as I now have to fix the DAG doc generation tests :)

File: catalog/utilities/dag_doc_gen/dag_doc_generation.py [hunk 1]
```diff
@@ -124,7 +124,10 @@ def get_dags_info(dags: DagMapping) -> list[DagInfo]:
         if doc:
             doc = fix_headings(doc)
 
-        dated = dag.catchup
+        provider_workflow = provider_workflows.get(dag_id)
+        dated = (
+            provider_workflow.dated if provider_workflow is not None else dag.catchup
+        )
```
Comment (AetherUnbound) [inline catalog/utilities/dag_doc_gen/dag_doc_generation.py:130]: Great catch!

**Review by AetherUnbound — APPROVED** [review]
Thanks for doing this! This is such a small thing that's a tremendous annoyance when testing locally 😄 

For others testing this, you can run `just catalog/ipython` and then the following to see what your calculated catchup values are for all DAGs:

```python
from airflow.models import DagBag
from pathlib import Path
dagbag = DagBag(dag_folder=Path(".").absolute(), include_examples=False)
print("<Dag>: <Catchup>")
for dag_id, dag in dagbag.dags.items():
    print(f"{dag_id}: {dag.catchup}")
```

**Pull request merged**